panic in pubsub lib Topic

Running `just e2e-frontend` in CI is getting a panic here: https://github.com/TBD54566975/ftl/actions/runs/10622270938/job/29446142300#step:6:1238


This command is running `ftl dev --recreate` under the hood.

```
panic: ack timeout

goroutine 180 [running]:
github.com/alecthomas/types/pubsub.(*Topic[...]).run(0x2203100)
	/home/runner/go/pkg/mod/github.com/alecthomas/types@v0.16.0/pubsub/pubsub.go:197 +0x6cb
created by github.com/alecthomas/types/pubsub.New[...] in goroutine 127
/home/runner/go/pkg/mod/github.com/alecthomas/types@v0.16.0/pubsub/pubsub.go:67
```

Note: this is not necessarily part of FTL's pubsub feature. It is caused by something using @alecthomass pubsub library

This happened in production at Aug 29, 2024 at 2:10:40.517 pm AEST
This happened 38s after the first occurence of this issue on the same controller: https://github.com/TBD54566975/ftl/issues/2539
The second time we saw that other issue, it was was not followed by this issue. Possibly because they are unrelated, or because that controller's pubsub set up was broken so subsequent issues didn't cause a similar panic.

production logs:
```
panic: ack timeout
goroutine 31 [running]:
github.com/alecthomas/types/pubsub.(*Topic[...]).run(0x1cf3dc0)
/root/go/pkg/mod/github.com/alecthomas/types@v0.16.0/pubsub/pubsub.go:197 +0x6cb
created by github.com/alecthomas/types/pubsub.New[...] in goroutine 1
/root/go/pkg/mod/github.com/alecthomas/types@v0.16.0/pubsub/pubsub.go:67 +0x10e
```

--- a/backend/controller/controller.go
+++ b/backend/controller/controller.go
@@ -1856,7 +1856,7 @@ func (s *Service) watchModuleChanges(ctx context.Context, sendChange func(respon
 		return err
 	}
 	initialCount := len(seedDeployments)
-	deploymentChanges := make(chan dal.DeploymentNotification, len(seedDeployments))
+	deploymentChanges := make(chan dal.DeploymentNotification, 32)
 	for _, deployment := range seedDeployments {
 		deploymentChanges <- dal.DeploymentNotification{Message: optional.Some(deployment)}
 	}

